internal/pkg/gateway/commit/commitadapter.go

```diff
@@ -28,13 +28,28 @@ func (adapter *PeerAdapter) CommitNotifications(done <-chan struct{}, channelNam
 	return channel.Ledger().CommitNotificationsChannel(done)
 }
 
-func (adapter *PeerAdapter) TransactionStatus(channelName string, transactionID string) (peerproto.TxValidationCode, error) {
+func (adapter *PeerAdapter) TransactionStatus(channelName string, transactionID string) (peerproto.TxValidationCode, uint64, error) {
 	channel, err := adapter.channel(channelName)
 	if err != nil {
-		return 0, err
+		return 0, 0, err
 	}
 
-	return channel.Ledger().GetTxValidationCodeByTxID(transactionID)
+	ledger := channel.Ledger()
+
+	status, err := ledger.GetTxValidationCodeByTxID(transactionID)
+	if err != nil {
+		return 0, 0, err
+	}
+
+	block, err := ledger.GetBlockByTxID(transactionID)
```

Comment:
Logging and debugging are the primary uses. It might also be useful as a constraint for future interactions, since we know the minimum block height required to see this transaction, but the Gateway is not making use of this today so not a showstopper.

Being able to grab just the block number out of the ledger index would be great, either along with the transaction ID or separately. If that's not practical then I'll give up on including the block number along with the validation code so I can make some progress on chaincode eventing, which also requires changes to this code.

Reply:
Yes, I was thinking on the lines of enhancing `GetTxValidationCodeByTxID` to return the block number along with the validation code. However, your PR will have to wait for that change. So one option could be to do that outside this PR. I leave the choice to you. If you prefer to do in this PR, I can spend some time on enhancing the above API

Reply:
Now using the new implementation to get block number directly from the `GetTxValidationCodeByTxID` call